prompting/rewards/multi_choice.py

```diff
@@ -0,0 +1,49 @@
+import re
+import time
+
+import numpy as np
+
+from prompting.base.dendrite import DendriteResponseEvent
+from prompting.rewards.reward import BaseRewardModel, BatchRewardOutput
+
+
+class MultiChoiceRewardModel(BaseRewardModel):
+    choices: tuple[str, str, str, str] = ("A", "B", "C", "D")
+
+    @property
+    def name(self) -> str:
+        return "multiple_choice"
+
+    def reward(self, reference: str, response_event: DendriteResponseEvent) -> BatchRewardOutput:
+        """Compute difference scores given a completion and reference pair."""
+        rewards = []
+        timings = []
+        classes = self.choices
+        completions: list[str] = response_event.completions
+
+        for completion in completions:
+            t0 = time.perf_counter()
+            # Step through the words in the completion and check if the reference is in the completion.
+            # Remove any punctuation too.
+            matches = [
+                word
+                for word in re.sub(r"\W", " ", completion).split()
+                if word in classes
+            ]
+
```

Comment:
I really believe we should simply force miners to respond in a fixed format. I'm alright with pushing this through now since this is what we tested, but let's make sure next week we modify this to be strict on the answer format:)

Reply:
It was made on purpose, to be the same format as MMLU